darc/darc.go

```diff
@@ -628,42 +627,64 @@ func EvalExprDarc(expr expression.Expr, getDarc GetDarc, acceptDarc bool, ids ..
 			if acceptDarc && found {
 				return true
 			}
+			// prevent cycles by checking the visited map
+			if _, ok := visited[s]; ok {
+				issue = errors.New("cycle detected")
+				return false
+			}
+			newVisited := make(map[string]bool)
```

Comment:
Is it actually necessary to copy the map ? If this is executed synchronously you can reuse the map in the recursions I think.

Reply:
the evaluation will expand every darc identity, suppose we have `darc:a && darc:b`, then the sign expr for both of them are are `darc:c`, then sign expr for `darc:c` is a regular ed25519 public key. in this situation, when evaluating `darc:a && darc:b`, the visited list will contain `darc:a, darc:c` when evaluating the LHS of `&&`. to evaluate the RHS of `&&`, the function will expand `darc:b` and find `darc:c`. if we're not copying the map, the evaluation will fail because `darc:c` is already in the visited list. if we do the copying, the visited list that is passed into `darc:b` expansion will be empty.

in summary, when the darc expansion is on different paths, we don't want to unify the visited list, copying fixes this problem.

I will write a test case for this scenario.